client/src/components/users/Users.integrate.test.jsx

```diff
@@ -0,0 +1,85 @@
+import {
+  render,
+  screen,
+  fireEvent,
+  waitFor,
+  act,
+} from "@testing-library/react";
+//User-event dispatches the events like they would happen if a user interacted with the document.
+
+import Users from "./Users";
+import * as api from "../../api";
+
+let getAllUserSpy;
+// we want to run a function or some other code repeatedly “before each” test that code can be put in the beforeEach function.
+beforeEach(() => {
+  // spyOn need property
+  getAllUserSpy = jest.spyOn(api, "getUsersApi").mockResolvedValue([]);
+});
+
+describe("render", () => {
+  it("loads all the users", async () => {
+    getAllUserSpy.mockResolvedValue([
+      {
+        name: "Jane Doe",
+        email: "Jane@gmail.com",
+        id: 42,
+      },
+    ]);
+
+    render(<Users />);
+    expect(getAllUserSpy).toHaveBeenCalled();
+    await waitFor(() => screen.findByText("Jane Doe"));
+    await expect(await screen.findByText("Jane@gmail.com")).toBeInTheDocument();
```

Comment:
`expect` isn't a promise so you don't need to await it.

```suggestion
    expect(await screen.findByText("Jane@gmail.com")).toBeInTheDocument();
```